src/analyzing/AST_to_IL.ml

```diff
@@ -363,6 +363,40 @@ and pattern_assign_statements env ?(eorig = NoOrig) exp pat =
   with
   | Fixme (kind, any_generic) -> fixme_stmt kind any_generic
 
+(*****************************************************************************)
+(* Exceptions *)
+(*****************************************************************************)
+and try_catch_else_finally env ~try_st ~catches ~opt_else ~opt_finally =
+  let try_stmt = stmt env try_st in
+  let catches_stmt_rev =
+    List.fold_left
+      (fun acc (ctok, exn, catch_st) ->
+        (* TODO: Handle exn properly. *)
+        let name = fresh_var env ctok in
+        let todo_pattern = fixme_stmt ToDo (G.Ce exn) in
+        let catch_stmt = stmt env catch_st in
+        (name, todo_pattern @ catch_stmt) :: acc)
+      [] catches
+  in
+  let all_handlers_rev =
+    match opt_else with
+    | None -> catches_stmt_rev
+    | Some (tok, else_st) ->
+        let name = fresh_var env tok in
+        let todo_pattern =
+          fixme_stmt ToDo (G.Ce (G.OtherCatch (("else_catch", tok), [])))
```

Comment:
This OtherCatch, just like OtherStmtWithStmt, OtherExpr, etc. are really last resort constructs when we're lazy and we don't want to properly handle the thing, especially if it's for translating a construct that seem used for only one language. Maybe here there's the opportunity to improve the AST_generic.catch_exn type and add indeed a CatchElse case, with a nice comment before explaining maybe its semantic, the difference between else and finally, etc.
Otherwise by using OtherCatch with those string like "else_catch" is not super type-safe because it's strongly coupled with the string we generate in ruby_to_generic.ml, which if we change one day the typechecker will not tell us to also update the code in AST_to_IL.ml

Reply:
I agree overall with Yoann, BUT if we add a proper test in this PR then we'll know if this breaks. And we could leave improving `AST_generic` for a separate PR perhaps.

Reply:
Wait, this is just a `TODO` IL node that you are generating, so I this won't break anything anyways.

Reply:
I am just no sure why we use `(G.OtherCatch (("else_catch", tok), []))` for the TODO node. That is not how the `else` is encoded by the Ruby parser anyways. You could just put the token there, and use `fresh_var ~str:"_try_else"` so that if someone examines the IL easily knows what that TODO corresponds to.

Reply:
Went with the unified `Try` statement direction. Marking this as resolved.